src/operator/nn/dropout-inl.h

```diff
@@ -253,7 +253,16 @@ class DropoutOp {
                            const TBlob &mask,
                            const TBlob &out) {
       Stream<xpu> *s = ctx.get_stream<xpu>();
-
+      Random<xpu, unsigned> *prnd = ctx.requested[1].get_random<xpu, unsigned>(s);
+      Tensor<xpu, 1, char> workspace =
+        ctx.requested[2].get_space_typed<xpu, 1, char>(Shape1(1), s);
+      // slice workspace
+      char *workspace_ptr = workspace.dptr_;
+      Tensor<xpu, 1, unsigned> random_number =
+        Tensor<xpu, 1, unsigned>(reinterpret_cast<unsigned *>(workspace_ptr),
+                                 Shape1(1), s);
+      prnd->GetRandInt(random_number);
+      uint64_t seed_ = 17 + reinterpret_cast<uint64_t>(&random_number[0]) % 4096;
```

Comment:
Why does it need the modulus operator `%` ? the modulus operator makes the `seed_` between 0+17~4096+17

Reply:
I'm just keeping the original logic here:
https://github.com/roywei/incubator-mxnet/blob/master/src/operator/nn/dropout-inl.h#L95

and here:
https://github.com/roywei/incubator-mxnet/blob/master/src/operator/nn/dropout-inl.h#L491

while fixing seed to respect mxnet seed.

Reply:
I think it should be `uint64_t seed_ = 17 + static_cast<uint64_t>(random_number[0]) % 4096;`, because the type of `random_number[0]` is `unsigned`.

https://github.com/apache/incubator-mxnet/blob/master/3rdparty/mshadow/mshadow/tensor.h#L591

Reply:
this will give a segfault during dropout. Also why would dropout on mult-gpu return different result? I thought the seed in fixed at global? so dropout on different GPU will use the same seed, thus return the same result?

Reply:
Sorry that I didn't express it clearly. If different GPUs use the same seed, the result of drouout of different GPU should be the same. When training a model, do GPUs use different random seed?